include/tscpp/util/TsSharedMutex.h

```diff
@@ -53,6 +54,36 @@
 
 namespace ts
 {
+class Strerror
+{
+public:
+  Strerror(int err_num)
+  {
+    _c_str = strerror_r(err_num, _buf, 256);
+    if (!_c_str) {
+      _c_str = "strerror_r() call failed";
+    } else {
+      _buf[255] = '\0';
+    }
+  }
+
+  char const *
+  c_str() const
+  {
+    return (_c_str);
+  }
+
+private:
+  char _buf[256];
+  char const *_c_str;
+};
+
+#if F
+#error "F preprocessor symbol defined"
+#endif
+
+#define F(FUNC_NAME, PTR, ERR_NUM) TSFatal("%s(%p) failed: %s (%d)", (FUNC_NAME), (PTR), Strerror(ERR_NUM).c_str(), (ERR_NUM));
```

Comment:
Why have a macro instead of an inline function?

Reply:
Brain fart, this review comment is legally elder abuse.